lib/middleware/request-handler.ts

```diff
@@ -119,66 +119,50 @@ export function elevateRequest (req: Interface.IOurRequest,
                                 next: restify.Next): void
 {
     var chunkIndex: number = 0;
-    var chunk: string; // the current chunk string is assembled here
-    var needComma: boolean = false; // need to append a ',' before adding more data
 
-    // returns a promise
-    function prepareResponse (): Promise<any> {
-        var resultP: Promise<any> = undefined;
+    // This is called each time something
+    function prepareResponseText(isEnd: boolean): string {
+        var text: string;
         if (++chunkIndex === 1) {
+            // This is the first time we're adding response text. Set the headers as well.
+            res.statusCode = 200;
             res.setHeader('content-type', 'application/json');
-            chunk = '{';
-            needComma = false;
+            if (!isEnd) {
+                // This is the beginning of a response with data.
+                text = '{"data":[';
+            } else {
+                // This is the beginning of a response without data.
+                text = '{';
+            }
         } else {
-            chunk = '';
+            if (!isEnd) {
+                // This is the continuation of a response that already has data and will add more.
+                text = ',';
+            } else {
+                // This is the continuation of a response that has data and will not add more.
+                text = '],';
+            }
         }
-
-        return resultP || Promise.resolve(undefined);
-    };
+        return text;
+    }
 
     function stringifyPair ( key: string, value: any ): string {
         return '"' + key.replace(/'/g, '\\$&') + '":' + JSON.stringify(value);
     }
 
     res.sendChunk = (data: any): Promise<void> => {
```

Comment:
Created #87 to track this -- I agree but subscriptions currently relies on this behavior and wasn't comfortable making that change all at once. I can tackle it after this one is in.